infrastructure/src/main/java/org/corfudb/infrastructure/Rank.java

```diff
@@ -26,16 +32,46 @@
      * Compares only the ranks. Does not use clientIds in the comparison.
      *
      * @param other rank to compare against
-     * @return True if this rank is lower than or equal to other rank,
-     *     False otherwise or if other is null
+     * @return True if this rank is lower than the other rank,
+     *         False otherwise or if other is null
      */
-    public boolean lessThanEqualTo(Rank other) {
+    public boolean lessThan(Rank other) {
         if (other == null) {
             return false;
         }
-        if (this == other) {
+
+        return this.rank.compareTo(other.rank) < 0;
+    }
+
+    /**
+     * CCompares only the ranks. Does not use clientIds in the comparison.
+     *
+     * @param other rank to compare against
+     * @return True if this rank is greater than or equal to the other rank,
+     *         False otherwise or if other is null
+     */
+    public boolean greaterThan(Rank other) {
+        return this.rank.compareTo(other.rank) > 0;
+    }
+
+    /**
+     * Compare must use clientId (PID) during equality comparison.
+     *
+     * @param other rank to compare against
+     * @return True if this rank is greater than or equal to the other rank,
+     *         False otherwise or if other is null
+     */
+    public boolean greaterThanOrEqual(Rank other) {
```

Comment:
should we make Rank comparable? I mean Rank implements Comparable interface

Reply:
When doing < or > you must exclude the client ID, when doing == the client ID must be used. So, I'm not sure Comparable would help here.